script.js

```diff
@@ -389,7 +389,8 @@ window.GaiaAppIcon = (function(exports) {
   };
 
   proto._relayout = function() {
-    this._size = this.clientWidth;
+    var bookmarkIcon = this.bookmark && this.bookmark.icon;
+    this._size = this.clientWidth || (bookmarkIcon && bookmarkIcon.size);
```

Comment:
ah, I see - why don't we use the component instead of taking its background image? The problem is this will break that optimisation for bookmarks when the component is being used normally.